lib/Ray/sources/Model/ModelAnimations.hpp

```diff
@@ -27,15 +27,24 @@ namespace RAY {
 			//! @brief Default constructor
 			~ModelAnimations() = default;
 
+			//! @brief Default move ctor
+			ModelAnimations(ModelAnimations &&) = default;
+
 			//! @brief Only single entity can hold these animations pointers
 			ModelAnimations &operator=(const ModelAnimations &) = delete;
 
 			//! @brief Castin Object to raw model animation pointer
 			ModelAnimation &operator[](int index);
 
+			//! @brief Same usage as the operator[] but const
```

Comment:
You can do a operator[] const for this purpose.

Reply:
nope operator[] is  should not be const